node_modules/gh-auth/tests/test-admins.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2014 "Fronteer LTD"
+ * Grasshopper Event Engine
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var assert = require('assert');
+var TestsUtil = require('gh-tests');
+var UsersTestsUtil = require('gh-users/tests/util');
+
+var AuthTestsUtil = require('gh-auth/tests/util');
+
+describe('Local admin authentication', function() {
+
+    describe('Login', function() {
+
+        /**
+         * Test that verifies that an admin can log in
+         */
+        it('verify an admin can log in', function(callback) {
+            // Let a global admin create a user
+            TestsUtil.getAnonymousGlobalAdminRestClient(function(anonymousClient) {
+                // The anonymous client should be able to log in
+                AuthTestsUtil.assertLogin(anonymousClient, 'administrator', 'administrator', callback);
+            });
+        });
+
+        /**
+         * Test that verifies validation when logging in
+         */
+        it('verify validation when logging in', function(callback) {
+            // Get an anonymous client
+            TestsUtil.getAnonymousGlobalAdminRestClient(function(anonymousClient) {
+
+                // Using bad credentials should not allow a user to get authenticated
+                AuthTestsUtil.assertLoginFails(anonymousClient, 'administrator', null, function() {
+                    AuthTestsUtil.assertLoginFails(anonymousClient, null, 'administrator', function() {
```

Comment:
Should we also verify that the admin credentials cannot be used to sign into an app?